File: c-sharp-design-patterns/OOP Principles/Coupling/LooseOrder.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace c_sharp_design_patterns.OOP_Principles.Coupling
{
    public class LooseOrder
    {

        private readonly INotificationService notificationService;

        public LooseOrder(INotificationService notificationService)
        {
            this.notificationService = notificationService;
        }

        public void PlaceOrder()
        {
            notificationService.SendNotification("Your order has been placed.");
        }
    }
}

File: OOP_Principles/Coupling/LooseOrder.java
package OOP_Principles.Coupling;

public class LooseOrder {

  private final NotificationService notificationService;

  public LooseOrder(NotificationService notificationService) {
    this.notificationService = notificationService;
  }

  public void PlaceOrder() {
    this.notificationService.SendNotification("Your order has been placed.");
  }
}
